### ISSUE TITLE
IDE0306 triggers for invalid case and the fix generate wrong code

### ISSUE THREAD

**Issue by wiz0u** [dotnet/roslyn#77596]
### Analyzer

IDE0306 Collection initialization can be simplified
triggers for code such as:
```csharp
var array = new[] { 1, 2, 3 };
Dictionary<int, int> dict;
Queue<int> queue;
// IDE0306 on those lines:
dict = new(array.Select(i => KeyValuePair.Create(i, i)));
queue = new(array);
```

For both cases, applying the suggested fix creates incorrect code with the [.. notation]

Expected: IDE0306 doesn't trigger on such lines (when there is no Add method?)

**Comment by CyrusNajmabadi** [comment]
i believe this was fixed.   will verify.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/76684
